### DISCUSSION TITLE
Functionality of "Create Service"

### DISCUSSION THREAD

**Discussion by oo-dev17** [winsiderss/systeminformer#756] [Questions / Answers] [answered]
Hi, first a BIG THANK YOU to the devs and supporters of this great software!

Since I found no documentation about the "Create Service" dialog and being unsuccessful using it, I want to ask:

Is this Dialog 'converting' an regular application exe into a windows service? (e.g. that responds to OnStart and OnStop commands)
Or asking different: Does this function mimic the NSSM (non sucking service manager)?

Thanks in advance!
oo_dev17

**Comment by dmex** [comment, marked as answer]
> Is this Dialog 'converting' an regular application exe into a windows service?

The dialog simply creates a new service entry in the SCM (service control manager) with the path to that executable. That application must support the [StartServiceCtrlDispatcher](https://docs.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicectrldispatcherw) function to successfully execute as a service using that dialog.

Most applications don't use the StartServiceCtrlDispatcher function so you have to instead use the Hacker menu > Run as... menu to create processes as a service rather than using the Tools > create service menu. You can also right-click any service on the Processes tab and use the Miscellaneous > Run as this user... menu to create processes as that service.
